lib/logstash/inputs/kinesis.rb

```diff
@@ -78,6 +81,15 @@ def register
     else
       creds = com.amazonaws.auth::DefaultAWSCredentialsProviderChain.new
     end
+
+    # If a role ARN is set then assume the role as a new layer over the credentials already created
+    unless @role_arn.nil?
+      session_id = "worker" + worker_id
```

Comment:
Is it possible to add an extra configuration parameter `role_session_name` for the sake of consistency with other logstash aws plugins ([this is in the common aws mixin gem shared by most aws plugins](https://github.com/logstash-plugins/logstash-mixin-aws/blob/f23c0bf56db5cc4bf4a4b07c6d618b94a5c83e94/lib/logstash/plugin_mixins/aws_config/generic.rb#L38))

Reply:
Useful to know. Will look at this today and see what the best approach is.